shell/platform/windows/public/flutter_windows.h

```diff
@@ -0,0 +1,168 @@
+// valid until FlutterDesktopRunWindowLoop has been called and returned.
+// Note that calling FlutterDesktopCreateWindow without later calling
+// FlutterDesktopRunWindowLoop on the returned reference is a memory leak.
+FLUTTER_EXPORT FlutterDesktopWindowControllerRef
+FlutterDesktopCreateWindow(int initial_width,
+                           int initial_height,
+                           const char* title,
+                           const char* assets_path,
+                           const char* icu_data_path,
+                           const char** arguments,
+                           size_t argument_count);
+
+// Shuts down the engine instance associated with |controller|, and cleans up
+// associated state.
+//
+// |controller| is no longer valid after this call.
+FLUTTER_EXPORT void FlutterDesktopDestroyWindow(
+    FlutterDesktopWindowControllerRef controller);
+
+// Loops on Flutter window events until the window is closed.
+//
+// Once this function returns, |controller| is no longer valid, and must not be
+// be used again, as it calls FlutterDesktopDestroyWindow internally.
+//
+// TODO: Replace this with a method that allows running the runloop
+// incrementally.
+FLUTTER_EXPORT void FlutterDesktopRunWindowLoop(
+    FlutterDesktopWindowControllerRef controller);
+
+// Returns the window handle for the window associated with
+// FlutterDesktopWindowControllerRef.
+//
+// Its lifetime is the same as the |controller|'s.
+FLUTTER_EXPORT FlutterDesktopWindowRef
+FlutterDesktopGetWindow(FlutterDesktopWindowControllerRef controller);
+
+// Returns the plugin registrar handle for the plugin with the given name.
+//
+// The name must be unique across the application.
+FLUTTER_EXPORT FlutterDesktopPluginRegistrarRef
+FlutterDesktopGetPluginRegistrar(FlutterDesktopWindowControllerRef controller,
+                                 const char* plugin_name);
+
+// Enables or disables hover tracking.
+//
+// If hover is enabled, mouse movement will send hover events to the Flutter
+// engine, rather than only tracking the mouse while the button is pressed.
+// Defaults to on.
+FLUTTER_EXPORT void FlutterDesktopWindowSetHoverEnabled(
+    FlutterDesktopWindowRef flutter_window,
+    bool enabled);
+
+// Sets the displayed title for |flutter_window|.
+FLUTTER_EXPORT void FlutterDesktopWindowSetTitle(
+    FlutterDesktopWindowRef flutter_window,
+    const char* title);
+
+// Sets the displayed icon for |flutter_window|.
+//
+// The pixel format is 32-bit RGBA. The provided image data only needs to be
+// valid for the duration of the call to this method. Pass a nullptr to revert
+// to the default icon.
+FLUTTER_EXPORT void FlutterDesktopWindowSetIcon(
+    FlutterDesktopWindowRef flutter_window,
+    uint8_t* pixel_data,
+    int width,
+    int height);
+
+// Gets the position and size of |flutter_window| in screen coordinates.
+FLUTTER_EXPORT void FlutterDesktopWindowGetFrame(
```

Comment:
Several of these functions appear to be declared but never implemented.

Is there any reason they can't just all be replaced with something that returns the underlying HWND for now?

Reply:
Ignore.